network/book.go

```diff
@@ -177,42 +177,61 @@ func (s *SimpleBook) Failed(addr string) {
 	e.failure++
 }
 
-// Get will return the top priority peer out of all peers that are currently not active. It should
+// OrderedSample will return the top priority peer out of all peers that are currently not active. It should
 // represent the peer that is most likely to allow us to connect successfully.
-func (s *SimpleBook) Get() (string, error) {
+func (s *SimpleBook) OrderedSample(params ...int) ([]string, error) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 	entries := s.slice(false)
 	if len(entries) == 0 {
-		return "", errBookEmpty
+		return nil, errBookEmpty
 	}
 	sort.Sort(byPriority(entries))
-	e := entries[0]
-	e.active = true
-	return e.addr, nil
+
+	sampleSize := s.sampleSize
+	if len(params) > 0 {
+		sampleSize = params[0]
+	}
+	if len(entries) > sampleSize {
+	    entries = entries[:sampleSize]
+	}
+	
+	addrs := s.getAddresses(entries, sampleSize)
```

Comment:
I generally don't like to use such small helper functions for iteration related things in Go.
You will do them often, and it is kind of like writing a helper function instead of an if statement.